src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -0,0 +1,164 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as semver from 'semver';
+import { MessageItem, OpenDialogOptions, workspace } from 'vscode';
+import { UserCancelledError } from 'vscode-azureextensionui';
+import { Platform, PythonAlias, TemplateFilter } from "../../constants";
+import { ext } from '../../extensionVariables';
+import { validateFuncCoreToolsInstalled } from '../../funcCoreTools/validateFuncCoreToolsInstalled';
+import { localize } from "../../localize";
+import { getFuncExtensionSetting, updateWorkspaceSetting } from '../../ProjectSettings';
+import { cpUtils } from "../../utils/cpUtils";
+import { funcHostProblemMatcher, funcHostTaskId } from "./IProjectCreator";
+import { ScriptProjectCreatorBase } from './ScriptProjectCreatorBase';
+
+export class PythonProjectCreator extends ScriptProjectCreatorBase {
+    public readonly templateFilter: TemplateFilter = TemplateFilter.Verified;
+    private readonly pythonVenvPathSetting: string = 'python.venvRelativePath';
+    private pythonAlias: string;
+    public getLaunchJson(): {} {
+        // currently the Python extension launches Windows with type: python and MacOS with type: pythonExperimental
+        const launchType: string = process.platform === Platform.Windows ? 'python' : 'pythonExperimental';
+        return {
+            version: '0.2.0',
+            configurations: [
+                {
+                    name: localize('azFunc.attachToJavaScriptFunc', 'Attach to Python Functions'),
+                    type: launchType,
+                    request: 'attach',
+                    port: 9091,
+                    host: 'localhost',
+                    preLaunchTask: funcHostTaskId
+                }
+            ]
+        };
+    }
+
+    public async addNonVSCodeFiles(): Promise<void> {
+        const funcCoreRequired: string = localize('funcCoreRequired', 'You must have the Azure Functions Core Tools installed to create, debug, and local Python functions.');
+        if (!await validateFuncCoreToolsInstalled(true /* forcePrompt */, funcCoreRequired)) {
+            throw new UserCancelledError();
+        }
+        await this.validatePythonVersion();
+        await this.setVirtualEnviornment();
```

Comment:
At some point, can you make sure to test this scenario:
1. Create a python project
1. Push it to GitHub
1. Clone it to a fresh location on your local computer
1. Run project

I'm just curious to see if all the virtual environment and launch/tasks stuff works from a fresh clone. Also try creating a project on one OS and then cloning it to a different OS. I don't think you necessarily need to do all of this for this PR, but it definitely needs to be done before we release.